src/NHibernate.Test/Async/NHSpecificTest/GH1180/FixtureByCode.cs

```diff
@@ -100,46 +101,47 @@ public async Task DecimalTypesAsync()
 			using (var session = OpenSession())
 			using (var transaction = session.BeginTransaction())
 			{
-				await (session.SaveAsync(new Entity {Amount = 3.14m}));
-				await (session.SaveAsync(new Entity {Amount = 42.13m}));
-				await (session.SaveAsync(new Entity {Amount = 17.99m}));
+				await (session.SaveAsync(new Entity {Amount = 3.141m}));
+				await (session.SaveAsync(new Entity {Amount = 42.131m}));
+				await (session.SaveAsync(new Entity {Amount = 17.991m}));
 
 				await (transaction.CommitAsync());
 			}
 
-			// whenTrue is constant, whenFalse is property -> works even before the fix
+			// whenTrue is constant, whenFalse is property
 			using (var session = OpenSession())
 			{
 				ICriteria tagCriteria = session.CreateCriteria(typeof(Entity));
 
 				var conditionalProjection = Projections.Conditional(
 					Restrictions.Not(
 						Restrictions.Ge(nameof(Entity.Amount), 20m)),
-					Projections.Constant(20m),
+					//Property scale is 2, make sure constant scale 3 is not lost
+					Projections.Constant(20.123m),
 					Projections.Property(nameof(Entity.Amount)));
 				tagCriteria.SetProjection(conditionalProjection);
 
 				// run query
 				var results = await (tagCriteria.ListAsync());
 
-				Assert.That(results, Is.EquivalentTo(new[] {20m, 42.13m, 20m}));
+				Assert.That(results, Is.EquivalentTo(new[] {20.123m, 42.13m, 20.123m}));
```

Comment:
Is that expected that SQLite saves Scale more than defined? I expected 42.131m to be saved as 42.13m according to mapping Scale value 2.

Reply:
SQLite does not have proper decimal type, it has `TestDialect.HasBrokenDecimalType` (not sure about the name of this property but you got the idea) set to true.